File: test/plugin/spring/fixtures/general/src/main/java/inga/client/StringLiteralHelper.java
package inga.client;

import org.springframework.http.HttpMethod;
import org.springframework.web.client.RestTemplate;

public class StringLiteralHelper {
    public static class WebClient {
        private final String path;
        private RestTemplate restTemplate;

        public WebClient(String path) {
           this.path = path;
        }

        public void get() {
            restTemplate.exchange(path, HttpMethod.GET, null, String.class);
        }
    }
}

File: test/plugin/spring/fixtures/general/src/main/kotlin/inga/client/StringLiteralHelper.kt
package inga.client

import org.springframework.http.HttpMethod
import org.springframework.web.client.RestTemplate

class StringLiteralHelper(
    val restTemplate: RestTemplate
) {
    @JvmInline
    value class WebClient(val path: String)

    fun WebClient.get() {
        restTemplate.exchange("${this.path}", HttpMethod.GET, null, String::class.java)
    }
    
    fun WebClient.post() {
        restTemplate.exchange("${this.path}", HttpMethod.POST, null, String::class.java)
    }
}
